modules/configuration/src/test/java/org/apache/ignite/internal/configuration/notifications/ConfigurationListenerTest.java

```diff
@@ -644,4 +667,366 @@ public void dataRace() throws Exception {
 
         assertEquals(List.of("deleted"), log);
     }
+
+    /** */
+    @Test
+    void testNoGetOrUpdateConfigValueForAny() throws Exception {
+        ChildConfiguration any0 = configuration.elements().any();
+
+        assertThrows(ConfigurationException.class, () -> any0.value());
+        assertThrows(ConfigurationException.class, () -> any0.change(doNothingConsumer()));
+
+        assertThrows(ConfigurationException.class, () -> any0.str().value());
+        assertThrows(ConfigurationException.class, () -> any0.str().update(""));
+
+        assertThrows(ConfigurationException.class, () -> any0.child2().value());
+        assertThrows(ConfigurationException.class, () -> any0.child2().change(doNothingConsumer()));
+
+        assertThrows(ConfigurationException.class, () -> any0.child2().i().value());
+        assertThrows(ConfigurationException.class, () -> any0.child2().i().update(100));
+
+        assertThrows(ConfigurationException.class, () -> any0.elements2().value());
+        assertThrows(ConfigurationException.class, () -> any0.elements2().change(doNothingConsumer()));
+        assertThrows(ConfigurationException.class, () -> any0.elements2().get("test"));
+
+        Child2Configuration any1 = any0.elements2().any();
+
+        assertThrows(ConfigurationException.class, () -> any1.value());
+        assertThrows(ConfigurationException.class, () -> any1.change(doNothingConsumer()));
+
+        assertThrows(ConfigurationException.class, () -> any1.i().value());
+        assertThrows(ConfigurationException.class, () -> any1.i().update(200));
+
+        configuration.elements().change(c0 -> c0.create("test", c1 -> c1.changeStr("foo"))).get(1, SECONDS);
+
+        Child2Configuration any2 = configuration.elements().get("test").elements2().any();
+
+        assertThrows(ConfigurationException.class, () -> any2.value());
+        assertThrows(ConfigurationException.class, () -> any2.change(doNothingConsumer()));
+
+        assertThrows(ConfigurationException.class, () -> any2.i().value());
+        assertThrows(ConfigurationException.class, () -> any2.i().update(300));
+    }
+
+    /** */
+    @Test
+    void testAnyListeners() throws Exception {
```

Comment:
Test is huge, maybe we should split it. Like "create" should be tested separately from "delete" and so on